packages/opentelemetry-plugin-xml-http-request/src/xhr.ts

```diff
@@ -0,0 +1,500 @@
+      return;
+    }
+
+    const resources: PerformanceResourceTiming[] = otperformance.getEntriesByType(
+      'resource'
+    ) as PerformanceResourceTiming[];
+
+    const resource = getResource(
+      spanUrl,
+      startTime,
+      resources,
+      this._usedResources,
+      maybeCors
+    );
+
+    if (resource.mainRequest) {
+      const mainRequest = resource.mainRequest;
+      this._markResourceAsUsed(mainRequest);
+
+      const corsPreFlightRequest = resource.corsPreFlightRequest;
+      if (corsPreFlightRequest) {
+        this._addChildSpan(span, corsPreFlightRequest);
+        this._markResourceAsUsed(mainRequest);
+      }
+      this._addSpanNetworkEvents(span, mainRequest);
+    }
+  }
+
+  /**
+   * Removes the previous information about span.
+   * This might happened when the same xhr is used again.
+   * @param xhr
+   * @private
+   */
+  private _cleanPreviousSpanInformation(xhr: XMLHttpRequest) {
+    const xhrMem = this._xhrMem.get(xhr);
+    if (xhrMem) {
+      const callbackToRemoveEvents = xhrMem.callbackToRemoveEvents;
+      if (callbackToRemoveEvents) {
+        callbackToRemoveEvents();
+      }
+      this._xhrMem.delete(xhr);
+    }
+  }
+
+  /**
+   * Creates a new span when method "open" is called
+   * @param xhr
+   * @param url
+   * @param method
+   * @private
+   */
+  private _createSpan(
+    xhr: XMLHttpRequest,
+    url: string,
+    method: string
+  ): types.Span | undefined {
+    if (isUrlIgnored(url, this._config.ignoreUrls)) {
+      this._logger.debug('ignoring span as url matches ignored url');
+      return;
+    }
+
+    const currentSpan = this._tracer.startSpan(url, {
+      parent: this._tracer.getCurrentSpan(),
+      attributes: {
+        [AttributeNames.COMPONENT]: this.component,
+        [AttributeNames.HTTP_METHOD]: method,
+        [AttributeNames.HTTP_URL]: url,
+      },
+    });
+
+    currentSpan.addEvent(EventNames.METHOD_OPEN);
+
+    this._cleanPreviousSpanInformation(xhr);
+
+    this._xhrMem.set(xhr, {
+      span: currentSpan,
+      spanUrl: url,
+    });
+
+    return currentSpan;
+  }
+
+  /**
+   * Marks certain [resource]{@link PerformanceResourceTiming} when information
+   * from this is used to add events to span.
+   * This is done to avoid reusing the same resource again for next span
+   * @param resource
+   * @private
+   */
+  private _markResourceAsUsed(resource: PerformanceResourceTiming) {
+    this._usedResources.add(resource);
+  }
+
+  /**
+   * Patches the method open
+   * @private
+   */
+  protected _patchOpen() {
+    return (original: OpenFunction): OpenFunction => {
+      const plugin = this;
+      return function patchOpen(
+        this: XMLHttpRequest,
+        method: string,
+        url: string,
+        async?: boolean,
+        user?: string | null,
+        pass?: string | null
+      ): void {
+        plugin._createSpan(this, url, method);
+
+        return original.call(this, method, url, true, user, pass);
```

Comment:
Creating a new thread on this line as the old one is deep in the history and on a line that has been outdated for a long time.

@draffensperger @mayurkale22

Reply:
Yeah, I think I missed this in my review, but I definitely agree that we should preserve the original behavior of the XHR function including the async parameter.